File: usb_android/android/src/main/kotlin/dev/yanshouwang/usb/MyUsbAccessory.kt
package dev.yanshouwang.usb

import android.hardware.usb.UsbAccessory

class MyUsbAccessory(private val collector: MyCollector) : MyUsbAccessoryHostApi {
    override fun getManufacturer(hashCode: Long): String {
        val usbAccessory = collector.instanceOf(hashCode) as UsbAccessory
        return usbAccessory.manufacturer
    }

    override fun getModel(hashCode: Long): String {
        val usbAccessory = collector.instanceOf(hashCode) as UsbAccessory
        return usbAccessory.model
    }

    override fun getDescription(hashCode: Long): String? {
        val usbAccessory = collector.instanceOf(hashCode) as UsbAccessory
        return usbAccessory.description
    }

    override fun getVersion(hashCode: Long): String? {
        val usbAccessory = collector.instanceOf(hashCode) as UsbAccessory
        return usbAccessory.version
    }

    override fun getUri(hashCode: Long): String? {
        val usbAccessory = collector.instanceOf(hashCode) as UsbAccessory
        return usbAccessory.uri
    }

    override fun getSerial(hashCode: Long): String? {
        val usbAccessory = collector.instanceOf(hashCode) as UsbAccessory
        return usbAccessory.serial
    }
}

File: usb_android/lib/src/my_usb_accessory.dart
import 'package:usb_platform_interface/usb_platform_interface.dart';

import 'my_api.dart';
import 'my_object.dart';

final myUsbAccessoryApi = MyUsbAccessoryHostApi();

class MyUsbAccessory extends MyObject implements UsbAccessory {
  MyUsbAccessory(super.hashCode);

  @override
  Future<String?> getDescription() {
    return myUsbAccessoryApi.getDescription(hashCode);
  }

  @override
  Future<String> getManufacturer() {
    return myUsbAccessoryApi.getManufacturer(hashCode);
  }

  @override
  Future<String> getModel() {
    return myUsbAccessoryApi.getModel(hashCode);
  }

  @override
  Future<String?> getSerial() {
    return myUsbAccessoryApi.getSerial(hashCode);
  }

  @override
  Future<String?> getUri() {
    return myUsbAccessoryApi.getUri(hashCode);
  }

  @override
  Future<String?> getVersion() {
    return myUsbAccessoryApi.getVersion(hashCode);
  }
}
